x-pack/plugins/gis/public/angular/map_controller.js

```diff
@@ -32,16 +34,39 @@ const REACT_ANCHOR_DOM_ELEMENT_ID = 'react-gis-root';
 
 const app = uiModules.get('app/gis', []);
 
-app.controller('GisMapController', ($scope, $route, config, kbnUrl) => {
+app.controller('GisMapController', ($scope, $route, config, kbnUrl, localStorage, AppState) => {
 
   const savedMap = $scope.map = $route.current.locals.map;
   let isDarkTheme;
   let unsubscribe;
 
   inspectorAdapters.requests.reset();
 
-  getStore().then(store => {
+  const $state = new AppState();
+  $scope.$listen($state, 'fetch_with_changes', function (diff) {
+    if (diff.includes('query')) {
+      $scope.updateQueryAndDispatch($state.query);
+    }
+  });
+  $scope.query = {};
+  $scope.indexPatterns = [];
+  $scope.updateQueryAndDispatch = function (newQuery) {
+    $scope.query = newQuery;
+    getStore().then(store => {
+      // ignore outdated query
+      if ($scope.query !== newQuery) {
```

Comment:
I don't understand the purpose for this 100%. Is this conditional right?

Reply:
`getStore` is async. `updateQueryAndDispatch` could be called before `getStore` returns. In that case the first call to `updateQueryAndDispatch` is garbage and just abort. The check compares newQuery to $scope.query. In the event that `updateQueryAndDispatch` is called before `getStore` returns then the objects will not be equal and the function just exits.